src/platform/forms-system/src/js/fields/FileField.jsx

```diff
@@ -334,6 +334,22 @@ class FileField extends React.Component {
               const deleteButtonText =
                 enableShortWorkflow && hasErrors ? 'Cancel' : 'Delete file';
 
+              const fileId = `file-name-${index}`;
+
+              const getUiSchema = innerUiSchema =>
+                typeof innerUiSchema === 'function'
+                  ? innerUiSchema({ fileId, index })
+                  : innerUiSchema;
+
+              // make index available to widgets in attachment ui schema
+              const indexedRegistry = {
+                ...registry,
+                formContext: {
+                  ...registry.formContext,
+                  pagePerItemIndex: index,
```

Comment:
I see that we're testing for `pagePerItemIndex` being there, but what does this do for us? `Ctrl + F` only finds 3 matches for it in the PR - one here and two in the tests.

Reply:
Some file uploads include a `attachmentSchema` or `attachmentName` which may include a component that checks for a `pagerPerItemIndex` which is undefined in this situation.